source/engines/bountyEngine.js

```diff
@@ -0,0 +1,35 @@
+const { Sequelize } = require("sequelize");
+const { Bounty } = require("../models/bounties/Bounty");
+const { userMention, Guild } = require("discord.js");
+const { listifyEN, congratulationBuilder } = require("../util/textUtil");
+
+/**
+ * @param {Guild} guild
+ * @param {Sequelize} database
+ * @param {Bounty} bounty
+ * @param {Company} company
+ * @param {string[]} completerIds
+ */
+function addCompleters(guild, database, bounty, company, completerIds) {
+	const rawCompletions = [];
+	for (const userId of completerIds) {
+		rawCompletions.push({
+			bountyId: bounty.id,
+			userId,
+			companyId: guild.id
+		})
+	}
+	database.models.Completion.bulkCreate(rawCompletions);
+	bounty.updatePosting(guild, company, database);
+	if (company.bountyBoardId) {
+		guild.channels.fetch(company.bountyBoardId).then(bountyBoard => {
+			return bountyBoard.threads.fetch(bounty.postingId);
+		}).then(posting => {
+			posting.send({ content: `${listifyEN(completerIds.map(id => userMention(id)))} ${completerIds.length === 1 ? "has" : "have"} been added as ${completerIds.length === 1 ? "a completer" : "completers"} of this bounty! ${congratulationBuilder()}!` });
+		});
+	}
+}
+
+module.exports = {
+	addCompleters
+}
```

Comment:
I'm not personally convinced by the "engines" naming convention for something like this. I'd personally expect engines to be fully self-sufficient (pure input to output) and closer to utility files with more internalized complexity, but the way that they are being used here is inherently not self-sufficient (needs a database object in order to function). I'm creating a `logic` folder instead, with the file name matching the simplest iteration on the associated functionality (so `stats.js` in my implemented case). What's your opinion on this?

Reply:
I don't have strong opinions on the name. I like the overall structure, but `stats.js` by itself is a little more narrow than my first instinct. Maybe groupings similar to our database models (Users, Seasons, Toasts, etc) would consolidate things intuitively?